TBufferMergerRootOutputer.cc

```diff
@@ -115,48 +120,82 @@ void TBufferMergerRootOutputer::write(unsigned int iLaneIndex) {
 	  }
 	}
 	if(lane.shouldWrite_) {
-	  lane.shouldWrite_=false;
-	  lane.file_->Write();
+          if(concurrentWrite_) {
+            writeWhenEnoughEvents(iLaneIndex);
+          } else {
+            auto group = iCallback.group();
+            queue_.push(*group, [this, iLaneIndex, callback=std::move(iCallback)]() mutable {
+                writeWhenEnoughEvents(iLaneIndex);
+                callback.doneWaiting();
+              });
+          }
 	}
       }
     });
+  lane.accumulatedFillTime_ += std::chrono::duration_cast<decltype(lane.accumulatedFillTime_)>(std::chrono::high_resolution_clock::now() - start);
+}
 
-  lane.accumulatedTime_ += std::chrono::duration_cast<decltype(lane.accumulatedTime_)>(std::chrono::high_resolution_clock::now() - start);
+void TBufferMergerRootOutputer::writeWhenBytesFull(unsigned int iLaneIndex) {
+  auto start = std::chrono::high_resolution_clock::now();
+  auto& lane = lanes_[iLaneIndex];
+  std::cout <<"lane "<< iLaneIndex<<" events since write "<<lane.nEventsSinceWrite_<<" "<<lane.nBytesWrittenSinceLastWrite_ << std::endl;
```

Comment:
This probably should be optional (rate of printing should be slow but non negligible)